patches/source/button_descriptions.c

```diff
@@ -0,0 +1,44 @@
+#include "button_descriptions.h"
+
+#include "menu.h"
+#include "attr.h"
+
+__attribute_reloc__ void (*update_element_alpha)(element_alpha_state_t* element_ptr, bool disabled);
+
+// Locations to branch back to, after the patched switch statement entry (used in button_descriptions_asm.s)
+__attribute_reloc__ void* after_update_button_text_jump_table;
+__attribute_reloc__ void* after_update_button_icons_jump_table;
+
+__attribute_reloc__ all_element_alphas_t* all_element_alphas;
+
+__attribute_used__ void update_gameplay_button_text() {
+    // Disable button descriptions used by the outer menu
+    update_element_alpha(&all_element_alphas->text.one_column.menu_selection, true);
+    update_element_alpha(&all_element_alphas->text.two_columns.left_cancel, true);
+    update_element_alpha(&all_element_alphas->text.two_columns.right_confirm, true);
+
+    // And also disable button descriptions used by other menus
+    update_element_alpha(&all_element_alphas->text.three_columns.left_change, true);
+    update_element_alpha(&all_element_alphas->text.three_columns.centre_finish, true);
+
+    // Show 'B: Cancel' on both the loader and start screens
+    update_element_alpha(&all_element_alphas->text.three_columns.centre_cancel, false);
```

Comment:
Oh, one difference from the original is that even 'B: Cancel' was conditional on *something* there, but I wasn't able to work out what that something was. I haven't replicated that condition here, but it could be worth double-checking if we should.

Reply:
I think I later found that this hid itself if the 'an error has occurred' message is displayed. I don't think I'll lose much sleep if we just don't replicate that 😛